src/controller/api.js

```diff
@@ -22,7 +23,34 @@ const extractHostName = (url) => {
 const config = async (body) => {
   const { rawMetadata, defaultRedirectUrl, redirectUrl, tenant, product } =
     body;
-  const idpMetadata = await saml.parseMetadataAsync(rawMetadata);
+
+  let idpMetadata;
+
+  if (!rawMetadata) {
+    throw new JacksonError('Please provide rawMetadata', 400);
+  }
+
+  if (!defaultRedirectUrl) {
+    throw new JacksonError('Please provide a defaultRedirectUrl', 400);
+  }
+
+  if (!redirectUrl) {
+    throw new JacksonError('Please provide redirectUrl', 400);
+  }
+
+  if (!tenant) {
+    throw new JacksonError('Please provide tenant', 400);
+  }
+
+  if (!product) {
+    throw new JacksonError('Please provide product', 400);
+  }
+
+  try {
```

Comment:
Probably not needed, the error should bubble up to the caller outside

Reply:
Noted.